backends/dpdk/dpdkArch.cpp

```diff
@@ -1086,10 +1096,30 @@ ConvertBinaryOperationTo2Params::postorder(IR::AssignmentStatement *a) {
         } else if (left->equiv(*r->right)) {
             IR::Operation_Binary *bin_expr;
             bin_expr = r->clone();
-            bin_expr->left = r->right;
-            bin_expr->right = r->left;
-            a->right = bin_expr;
-            return a;
+            if (isCommutativeBinaryOperation(r)) {
+                bin_expr->left = r->right;
+                bin_expr->right = r->left;
+                a->right = bin_expr;
+                return a;
+            } else {
+                // Expressions like "a = 10 - a" should be replaced with the following block
```

Comment:
will it work if we convert to `a = -a + 10`

Reply:
I don't think so. Negation is another unary operation and would need an assignment like tmp =-a (Dpdk does not support any complex operations)
Also, handling such a way will not handle the other non-commutative operations like shl, shr. We will then have to find inverse of all such operations which will unnecessarily complicate the implementation here.

Reply:
is p4 ir not in ssa form?

Reply:
No. I don't think so.